certora/specs/Gates.spec

```
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+using ERC20Standard as ERC20;
+using ERC20Helper as ERC20Helper;
+using MorphoMarketV1Adapter as MorphoMarketV1Adapter;
+using MorphoVaultV1Adapter as MorphoVaultV1Adapter;
+
+methods {
+    function multicall(bytes[]) external => NONDET DELETE;
```

```suggestion
    function multicall(bytes[]) external => HAVOC_ALL DELETE;
```
since it's not a view function, and it should go through as it shouldn't be called

---

Done in https://github.com/morpho-org/vault-v2/pull/620/commits/59a8d61de8f995cd921530c5d47ab8140e15abcd.

---

Reverted in https://github.com/morpho-org/vault-v2/pull/620/commits/c225e84583c8ff60e483dcb5bd32ec0220eeb3e8 as it makes the verification timeout, see [here](https://prover.certora.com/output/129828/69be18e9d8814fa4af6866c1f267f3ed?anonymousKey=6d4a40f3a15d3390ffce428729c8033be46f3c5f) and [here](https://prover.certora.com/output/129828/983433ff35d14dcbb63213a8563b8fa7?anonymousKey=cf048f7fa476e495fc9382794888ce8a55e079e1).